src/components/Contacts/ContactDetails/ContactDetailsTab/Tags/ContactTags.tsx

```diff
@@ -21,22 +25,37 @@ const ContactTagAddTagText = styled(Typography)(({ theme }) => ({
 }));
 
 interface ContactTagsProps {
+  accountListId: string;
   contactId: string;
-  contactTags: string[];
 }
 
 export const ContactTags: React.FC<ContactTagsProps> = ({
+  accountListId,
   contactId,
-  contactTags,
 }) => {
   const { t } = useTranslation();
+  const { data, loading } = useGetContactTagsQuery({
+    variables: {
+      accountListID: accountListId, // value for 'accountListID'
+      contactId: contactId, // value for 'contactId'
+    },
+  });
+
   const handleTagDelete = (tag: string) => {
-    const index = contactTags.indexOf(tag);
-    if (index > -1) {
-      contactTags = contactTags.splice(index, 1);
-      // TODO: add mutation to update tag for contact
-    }
-    console.info('You clicked the Delete tag for tag' + tag + '' + contactId);
+    console.info('You clicked the Delete tag for tag ' + tag + ' ' + contactId);
+    const list = data.contact.tagList;
+    const updatedTags: string[] = [];
+    list.forEach((element) => {
+      element != tag ? updatedTags.push(tag) : null;
+    });
```

Comment:
`.filter` would be better here. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter